File: 백준/Silver/10828. 스택/스택.py
import sys

def main() -> None:
    input = sys.stdin.readline

    N = int(input().strip()) # 명령 수
    stack = []
    
    for _ in range(N):
        command = input().strip().split()
        if command[0] == "push":
            stack.append(int(command[1]))
        elif command[0] == "pop":
            if stack:
                print(stack.pop())
            else:
                print(-1)
        elif command[0] == "size":
            print(len(stack))
        elif command[0] == "empty":
            if stack:
                print(0)
            else:
                print(1)
        elif command[0] == "top":
            if stack:
                print(stack[-1])
            else:
                print(-1)
if __name__ == "__main__":
    main()

File: archive/boj/10828/yangwj/스택.js
const fs = require("fs");
const filePath = process.platform === "linux" ? "/dev/stdin" : "input.txt";
const input = fs.readFileSync(filePath).toString().trim().split("\n");

const stack = [];
const result = []; 
const n = Number(input[0]);

for (let i = 1; i <= n; i++) {
  const command = input[i].split(" ");
  
  if (command[0] === "push") {
    stack.push(Number(command[1]));
  } 
  else if (command[0] === "pop") {
    result.push(stack.length === 0 ? -1 : stack.pop());
  } 
  else if (command[0] === "size") {
    result.push(stack.length);
  } 
  else if (command[0] === "empty") {
    result.push(stack.length === 0 ? 1 : 0);
  } 
  else if (command[0] === "top") {
    result.push(stack.length === 0 ? -1 : stack[stack.length - 1]);
  }
}

console.log(result.join("\n")); 